CodeStarterCamp_Week3/main.swift

```diff
@@ -8,51 +8,79 @@
 
 import Foundation
 
-enum Coffee {
+enum Coffee: CaseIterable {
     case americano
     case latte
     case espresso
+    
+    var price: Int {
+        switch self {
+        case .espresso:
+            return 3800
+        case .americano:
+            return 4100
+        case .latte:
+            return 4500
+        }
+    }
 }
 
-struct Person {
+class Person {
     let name: String
     var money: Int
     
-    mutating func buySomething(for price: Int) {
-        let hasEnoughMoney = self.money >= price
-        if hasEnoughMoney {
-            self.money -= price
-            print("\(self.name)이 구매를 완료했습니다. 잔액은 \(self.money)원 입니다")
-        } else {
-            print("잔액이 부족합니다.")
+    init(name: String, money: Int) {
+        self.name = name
+        self.money = money
+    }
+    
+    func buy(coffee: Coffee, from coffeeShop: CoffeeShop) {
+        if Coffee.allCases.contains(coffee) {
+            let hasEnoughMoney = self.money >= coffee.price
+            guard hasEnoughMoney else {
+                print("잔액이 \(coffee.price-self.money)원만큼 부족합니다.")
+                return
+            }
+            self.money -= coffee.price
+            print("결제가 완료되었습니다. 잔액은 \(self.money)원입니다.")
+            coffeeShop.takeOrder(coffee: coffee, from: self)
+        }
+        else {
```

Comment:
```suggestion
        } else {
```
`if-else` 에서 `else`는 보통 요렇게 올려씁니다!